### PR TITLE
Restricted public access

### PR DESCRIPTION
**Pull request by Bunny303**

### CODE REVIEW

File: apps/api/src/campaign/campaign.controller.ts [hunk 1]
```diff
@@ -28,7 +28,6 @@ export class CampaignController {
   ) {}
 
   @Get('list')
-  @Public()
```
Comment (kachar) [inline apps/api/src/campaign/campaign.controller.ts:31]: We need to leave this one public as it used on the main campaigns page.

File: apps/api/src/expenses/expenses.controller.ts [hunk 2]
```diff
@@ -11,12 +11,19 @@ export class ExpensesController {
   constructor(private readonly expensesService: ExpensesService) {}
 
   @Public()
```
Comment (kachar) [inline apps/api/src/expenses/expenses.controller.ts:13]: It should be either `@Public()` or `@Roles()`

I'm not certain about the behavior of both together.

```suggestion
```

File: apps/api/src/recurring-donation/recurring-donation.controller.ts [hunk 3]
```diff
@@ -11,19 +11,20 @@ export class RecurringDonationController {
...
+  @Roles({
+    roles: [RealmViewSupporters.role, ViewSupporters.role],
+    mode: RoleMatchingMode.ANY,
+  })
   findAll() {
     return this.recurringDonationService.findAll()
   }
 
   @Get(':id')
-  @Public()
```
Comment (kachar) [inline apps/api/src/recurring-donation/recurring-donation.controller.ts:26]: I don't find reasons to have this endpoint available to authenticated users. Maybe we can restrict it to the admins as well

File: apps/api/src/campaign/campaign.controller.ts (hunk 1 above)
Comment (Bunny303) [reply apps/api/src/campaign/campaign.controller.ts:31]: In the main campaign page should be returned only active and completed campaigns. I thought this route to be used for admin list with campaigns and new one to be created for the main page.

Comment (Bunny303) [reply apps/api/src/campaign/campaign.controller.ts:31]: Actually in the existing one should be added a filter for active and completed and new route with admin access only could be added.

**Review by kachar — APPROVED** [review]
Outstanding work @Bunny303  Now the api feels much more secure :shield: :closed_lock_with_key:

**Pull request merged**